tests/conftest.py

```diff
@@ -35,21 +35,33 @@ def __init__(self, id, labels, attrs: dict | None = None):  # noqa: A002
         self.id = id
         self.labels = labels
         self.status = "created"
+        self.run_duration = 0.5
+        self.start_time = time.time()
 
         if attrs:
             self.attrs = attrs
         else:
             self.attrs = {"State": {"ExitCode": 0}}
 
     def reload(self):
+        """Simulate reload the status of a Docker container.
+
+        Checking if a detached Container has completed is performed by running its
+        reload method.  This waits 0.1s, then only resolves to 'exited' if the container
+        instance has existed longer than self.run_duration, thereby simulating a total
+        running time of X seconds.
```

Comment:
This was also pretty critical for testing.

After we run a detached container -- real or mocked -- we immediately go into a loop and poll the `Container.reload()` method.  In a way, it's _this_ method that really drives how long our code thinks the container ran.  This mocked `Container` class allows for pretty easily simulating a Container running for `X` seconds, by having the `reload()` method only return `"exited"` after a given duration of time.